bonsai-json-eval/src/main/java/com/phonepe/commons/bonsai/json/eval/JsonPathFilterEvaluationEngine.java

```diff
@@ -199,17 +211,29 @@ public Boolean visit(GenericFilter filter) {
 
 
     private <T> Boolean applyAllMatchFilter(Filter filter, TypeRef<List<T>> typeRef, Predicate<T> predicate) {
-        List<T> nonNullValues = nonNullValues(filter, typeRef);
-        return isNotEmpty(nonNullValues) && nonNullValues.stream().allMatch(predicate);
+        List<T> values = readContextValues(filter, typeRef);
```

Comment:
```sh
Benchmark                   Mode  Cnt        Score        Error  Units
JsonPathEvalBenchmark.newerImpl  thrpt    5  4319391.869 ± 107293.083  ops/s
JsonPathEvalBenchmark.olderImpl  thrpt    5  3545470.314 ± 133759.198  ops/s
```

Reply:
This benchmark excludes the List->Set conversion from query-dsl. That would be added improvement